### PR TITLE
Links and Formatting Fixes for the Docs

### PR DESCRIPTION
**Pull request by jbouffard**
This PR adds links and fixes the formatting of the guides for GeoPySpark.

### CODE REVIEW

File: docs/guides/catalog.rst [hunk 1]
```diff
@@ -1,59 +1,56 @@
+Catalog
+=======
+
+The ``catalog`` module allows for users to retrieve information, query,
+and write to/from GeoTrellis layers.
+
+Before begining, all examples in this guide need the following, boilerplate
```
Comment (jpolchlo) [inline docs/guides/catalog.rst]: Comma between following and boilerplate should be removed

File: docs/guides/core-concepts.rst [hunk 2]
```diff
@@ -14,12 +7,21 @@ terminology and data representations have carried over. This section
 seeks to explain this jargon in addition to describing how GeoTrellis
 types are represented in GeoPySpark.
 
+Before begining, all examples in this guide need the following, boilerplate
```
Comment (jpolchlo) [inline docs/guides/core-concepts.rst]: Another errant comma after "following'.

File: docs/guides/layers.rst [hunk 3]
```diff
@@ -1,25 +1,26 @@
+Working With Layers
+===================
+
+Before begining, all examples in this guide need the following, boilerplate
```
Comment (jpolchlo) [inline docs/guides/layers.rst]: Attack of the bad comma continues.

File: docs/guides/layers.rst [hunk 4]
```diff
@@ -210,15 +215,15 @@ the ``layout`` given.
...
     raster_layer.collect_metadata(layout=layout_definition)
 
 Reproject
 ^^^^^^^^^
 
-``reproject`` will change the projection the rasters within the layer to
-the given ``target_crs``. This method does not sample past the tiles'
-boundaries.
+:meth:`~geopyspark.geotrellis.layer.RasterLayer.reproject` will change the
+projection the rasters within the layer to the given ``target_crs``. This
```
Comment (jpolchlo) [inline docs/guides/layers.rst]: "...projection _of_ the rasters..."

File: docs/guides/layers.rst [hunk 5]
```diff
@@ -481,11 +489,12 @@ pyramided. That is, we create a level-of-detail hierarchy that covers
 the same geographical extent, while each level of the pyramid uses one
 quarter as many pixels as the next level. This allows us to zoom in and
 out when the layer is being displayed without using extraneous detail.
-The ``pyramid`` method will produce an instance of ``Pyramid`` that will
-contain within it multiple ``TiledRasterLayer``\ s. Each layer
-corresponds to a zoom level, and the number of levels depends on the
-``zoom_level`` of the source layer. With the max zoom of the ``Pyramid``
-being the source layer's ``zoom_level``, and the lowest zoom being 0.
+The :meth:`~geopyspark.geotrellis.layer.TiledRasterLayer.pyramid` method
+will produce an instance of ``Pyramid`` that will contain within it multiple
```
Comment (jpolchlo) [inline docs/guides/layers.rst]: Should Pyramid and TiledRasterLayer be links here?  Not saying they certainly should.  Just a question, really.

File: docs/guides/map-algebra.rst [hunk 6]
```diff
@@ -52,9 +20,35 @@ roughly equivalent to a 2d convolution operation.
 and if a local operation requires multiple inputs, those inputs must
 have the same layout and projection.
 
-**Note**: Throughout this guide, this ``.lookup(0, 0)[0].cells`` is used
-on the resulting layer. This call simply retrieves the numpy array of
-the first tile within the layer.
+Before begining, all examples in this guide need the following, boilerplate
```
Comment (jpolchlo) [inline docs/guides/map-algebra.rst]: There's that comma again.  Yay for copy/paste!

File: docs/guides/layers.rst (hunk 5 above)
Comment (jbouffard) [reply docs/guides/layers.rst]: `TiledRasterLayer` was linked to earlier in the docs, so I didn't think it needed to be here. `Pyramid` should become a link.

**Pull request merged**